src/insights_client/tests/test_metrics.py

```diff
@@ -34,27 +35,92 @@ def factory(config):
+    rhsm_config_file = _tempfile("[nonsense]\ntest=test")
+    metrics_client = MetricsHTTPClient(config_file=config_file.name, rhsm_config_file=rhsm_config_file.name)
+    assert metrics_client.auth == ("testuser", "testpass")
+    assert not metrics_client.cert
+    assert metrics_client.base_url == "cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_default_cert_auth(proxy_settings, config_file_factory):
+    '''
+    Verify that when insights-client configuration is used, CERT auth is selected
+    if no authmethod is defined in the configuration file
+    '''
+    config_file = config_file_factory("username=testuser\npassword=testpass")
+    rhsm_config_file = _tempfile("[nonsense]\nauth=test")
+    metrics_client = MetricsHTTPClient(
+        config_file=config_file.name, rhsm_config_file=rhsm_config_file.name,
+        cert_file="/path/to/test/cert.pem", key_file="/path/to/test/key.pem")
+    assert not metrics_client.auth
+    assert metrics_client.cert == ("/path/to/test/cert.pem", "/path/to/test/key.pem")
+    assert metrics_client.base_url == "cert.cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_is_not_satellite(proxy_settings, config_file_factory, rhsm_config_file_factory):
+    '''
+    Verify that when the configured RHSM hostname matches one of the Red Hat subscription URLs,
+    it is determined to NOT be a Satellite-subscribed host
+    '''
+    config_file = config_file_factory("")
+    rhsm_config_file = rhsm_config_file_factory(hostname="subscription.rhsm.redhat.com")
+    metrics_client = MetricsHTTPClient(
+        config_file=config_file.name, rhsm_config_file=rhsm_config_file.name,
+        cert_file="/path/to/test/cert.pem", key_file="/path/to/test/key.pem")
+    assert not metrics_client.auth
+    assert metrics_client.cert == ("/path/to/test/cert.pem", "/path/to/test/key.pem")
+    assert metrics_client.base_url == "cert.cloud.redhat.com"
+    assert metrics_client.api_prefix == "/api"
+
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_is_not_satellite(proxy_settings, config_file_factory, rhsm_config_file_factory):
+    '''
+    Verify that when the configured RHSM hostname does NOT match one of the Red Hat subscription URLs,
+    it is determined to be a Satellite-subscribed host
+    '''
+    config_file = config_file_factory("")
+    rhsm_config_file = rhsm_config_file_factory(
+        hostname="satellite.example.com",
+        repo_ca_cert="/path/to/cacert/cert.pem", consumerCertDir="/path/to/sat/certs")
+    metrics_client = MetricsHTTPClient(
+        config_file=config_file.name, rhsm_config_file=rhsm_config_file.name)
+    assert not metrics_client.auth
+    assert metrics_client.cert == ("/path/to/sat/certs/cert.pem", "/path/to/sat/certs/key.pem")
+    assert metrics_client.port == "443"
```

Comment:
I am not very happy about how this is taken from the default value there in the fixture. All the other tested values are here in this test case.